src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1381,7 +1384,7 @@ private ArrayTypeSymbol VisitArrayInitializer(BoundArrayCreation node)
                 }
                 else
                 {
-                    elementType = TypeSymbolWithAnnotations.Create(bestType, BestTypeInferrer.GetNullableAnnotation(bestType, resultBuilder));
+                    elementType = TypeSymbolWithAnnotations.Create(bestType, BestTypeInferrer.GetNullableAnnotation(bestType, speakableResultBuilder));
```

Comment:
I think the conversion to a speakable annotation is supposed to occur at the end of type inference, not on the inputs.